components/eamxx/src/physics/nudging/atmosphere_nudging.cpp

```diff
@@ -177,26 +241,27 @@ void Nudging::update_time_step (const int time_s)
   if (time_s >= NudgingData_aft.time)
     {
       const int time_index = time_s/time_step_file;
-      data_input->read_variables(time_index-1);
-      Kokkos::deep_copy(NudgingData_bef.T_mid,T_mid_ext);
-      Kokkos::deep_copy(NudgingData_bef.p_mid,p_mid_ext);
-      Kokkos::deep_copy(NudgingData_bef.hw,horiz_winds_ext);
-      Kokkos::deep_copy(NudgingData_bef.qv,qv_ext);
-      NudgingData_bef.time = time_step_file*(time_index);
-
-      data_input->read_variables(time_index);
-      Kokkos::deep_copy(NudgingData_aft.T_mid,T_mid_ext);
-      Kokkos::deep_copy(NudgingData_aft.p_mid,p_mid_ext);
-      Kokkos::deep_copy(NudgingData_aft.hw,horiz_winds_ext);
-      Kokkos::deep_copy(NudgingData_aft.qv,qv_ext);
+      Kokkos::deep_copy(NudgingData_bef.T_mid,NudgingData_aft.T_mid);
```

Comment:
Instead of a deep copy, you could consider doing a swap

```
std::swap (NudgingData_bef,NudringData_aft);
```
This will simply swap a few pointers, without doing unnecessary deep copies.

Reply:
Note: all you need to do to use std::swap is to implement `NudgingData& operator=(const NudgingData&)` (which should do a shallow assignment).

Reply:
Actually, the compiler should already implicitly provide the correct one, but you may consider making it clear via
```
NudgingData& operator=(const NudgingData&) = default;
```
in the class declaration.  Explicit intent is always nice.

Reply:
This is nice, thanks. I have updated to do this